### PR TITLE
[NDM] Fix for ignored_ip_addresses in autodiscovery

### PR DESCRIPTION
**Pull request by a-rhodes**
### What does this PR do?
Adds fix and unit test for a bug that currently exists in the NDM Autodiscovery Listener.

### Motivation
When adding ignored_ip_addresses to the NDM Autodiscovery config in datadog.yaml, the SNMP Listener fails to start and no error is logged.

### Describe how to test/QA your changes
Unit test included in PR

### Possible Drawbacks / Trade-offs
None

### Additional Notes
Related issue: https://github.com/DataDog/datadog-agent/issues/30181
Unit test showing original unmarshal bug: https://github.com/DataDog/datadog-agent/pull/30177

### CODE REVIEW

**Comment by dustmop** [comment]
Hello @a-rhodes! Thank you for tracking down the root cause of this issue and supplying this bug fix! It looks good to me, I'm going to reach out to the codeowners to get the necessary approvals to merge it.

I pushed a commit with a release note [to the branch](https://github.com/DataDog/datadog-agent/commit/cd17867b8c704d093e3fa64e8fa952c6e916d378) `dustin.long/fix-config-autodiscovery`. It's required for passing our CI. Would you mind either cherry-picking that commit and pushing it to your branch, or [installing and running](https://github.com/DataDog/datadog-agent/blob/66de8311bec89875e3b80aa7a1383bb9cc1158ae/docs/public/guidelines/contributing.md#reno) `reno` yourself? Thanks!

**Review by davidor — APPROVED** [review]

File: pkg/config/autodiscovery/autodiscovery_test.go
Comment (L3n41c) [inline pkg/config/autodiscovery/autodiscovery_test.go:48]: This follows the same pattern as what is done above in the same file.
Yet, it is problematic as it won’t skip the evaluation of `configListeners[0]` when `configListeners` is empty.
Indeed, `assert.XXX(…)` will eventually make the whole test fail, but it doesn’t abort its execution prematurely.
On the other hand, `require.XXX(…)` makes the test abort prematurely and skips the execution of the rest of it.

So, in order to avoid panicking when `configListeners` is empty, we shoud either do:
```suggestion
	require.Len(t, configListeners, 1)
	assert.Equal(t, "snmp", configListeners[0].Name)
```

or, if we don’t want to skip the execution of the rest of the test,
```suggestion
	if assert.Len(t, configListeners, 1) {
		assert.Equal(t, "snmp", configListeners[0].Name)
	}
```

File: pkg/config/autodiscovery/autodiscovery_test.go
Comment (L3n41c) [inline pkg/config/autodiscovery/autodiscovery_test.go:61]:
```suggestion
	require.Len(t, configListeners, 1)
	assert.Equal(t, "snmp", configListeners[0].Name)
```

or

```suggestion
	if assert.Len(t, configListeners, 1) {
		assert.Equal(t, "snmp", configListeners[0].Name)
	}
```

**Review by jedupau — APPROVED** [review]

**Comment by a-rhodes** [comment]
Release note has been added and unit test has been updated to use `require`

**Review by rtrieu — APPROVED** [review]
LGTM

**Pull request merged**
